src/box/lua/load_cfg.lua

```diff
@@ -1149,7 +1149,24 @@ local function get_option_from_env(option)
 
     -- This code lean on the existing set of template_cfg
     -- types for simplicity.
-    if param_type:find('table') and raw_value:find(',') then
+    if param_type:find('table') and (raw_value:startswith('{') or
+                                     raw_value:startswith('[')) then
+        return json.decode(raw_value)
+    elseif param_type:find('table') and raw_value:find('=') then
+        assert(not param_type:find('boolean'))
+        local res = {}
+        for _, v in ipairs(raw_value:split(',')) do
+            local eq = v:find('=')
+            if eq == nil then
+                error(err_msg_fmt:format(env_var_name, option,
```

Comment:
There must be a test case for this error.

Reply:
It is quite hard to test error thrown from `box.cfg{}` in `luatest.server`, as it is being run during `server:start()` and not inside `server:exec()`. Generally, we can test it via `popen()`, but it would be quite ugly. You can see it in the latest test version.

Reply:
Perhaps it would be ugly, but without test it would be more ugly. If a code is not tested at all - it can simply be dropped during refactoring. To be sure I discussed it with colleagues. The conclusion is that there must be a test.

Reply:
How about `treegen`+`justrun`? See the example: https://github.com/tarantool/tarantool/blob/55295f5f1be769cb88ceb262d8817d2c9e34c32c/test/app-luatest/override_panic_test.lua.

Reply:
Added `popen` helper, it seems somewhat clean this way.